Backend/server.js

```diff
@@ -84,22 +84,11 @@ app.get('/getSubmission', async (req, res) => {
   }
 });
 
-app.get('/logout', limiter, async (req, res) => {
-  const user_id = req.query.user_id;
-  const login_time = req.query.login_time;  
-  console.log(login_time)
-  
-  let db = getDb();
-  db.updateOne({ "username": { $eq: user_id } }, { $set: { "lastLogin": login_time } })
-  console.log("< logged out user " + user_id)
-  res.status(200).json({ message: "Logged out!" });
-})
-
-
 
 app.post('/loginUser', limiter, async (req, res) => {
   const u = req.body.username;
   const p = req.body.password;
+  const apiKey = req.body.apiKey;
   
   if (!u || !p) {
     return res.status(400).json({message: "Username and password required!"});
```

Comment:
The entire server crashes if a single user tries to login without an API key saved. e.g. if you register an account then clear local storage/use a different device/browser to login later, the server will crash. We should make UI to ask for the API key again, but for now we should probably check if it's actually there or not before using it (like it does for username and password already).

This is also a problem in some other endpoints. The same guard should probably be added on those, too.

Reply:
What other routes you think would suffer this same issue?

Reply:
I fixed them in 79a4864